ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift

```diff
@@ -259,45 +259,92 @@ extension BrowserViewController: TopToolbarDelegate {
-          )
-          topToolbar.locationView.loading = tabManager.selectedTab?.loading ?? false
-          guard !Task.isCancelled else { return true }  // user pressed stop, or typed new url
-          switch result {
-          case .loadInterstitial(let service):
-            showWeb3ServiceInterstitialPage(service: service, originalURL: fixupURL)
+    }
+
+    if let url = URL(string: text), url.scheme == "brave" {
+      topToolbar.leaveOverlayMode()
+      return handleChromiumWebUIURL(url)
+    }
+
+    guard let fixupURL = URIFixup.getURL(text) else {
+      return false
+    }
+    // Do not allow users to enter URLs with the following schemes.
+    // Instead, submit them to the search engine like Chrome-iOS does.
+    if !["file"].contains(fixupURL.scheme) {
+      // check text is decentralized DNS supported domain
+      if let decentralizedDNSHelper = self.decentralizedDNSHelperFor(url: fixupURL) {
+        topToolbar.leaveOverlayMode()
+        updateToolbarCurrentURL(fixupURL)
+        topToolbar.locationView.loading = true
+        let result = await decentralizedDNSHelper.lookup(
+          domain: fixupURL.schemelessAbsoluteDisplayString
+        )
+        topToolbar.locationView.loading = tabManager.selectedTab?.loading ?? false
+        guard !Task.isCancelled else { return true }  // user pressed stop, or typed new url
+        switch result {
+        case .loadInterstitial(let service):
+          showWeb3ServiceInterstitialPage(service: service, originalURL: fixupURL)
+          return true
+        case .load(let resolvedURL):
+          if resolvedURL.isIPFSScheme {
+            return handleIPFSSchemeURL(resolvedURL)
+          } else {
+            finishEditingAndSubmit(resolvedURL)
             return true
-          case .load(let resolvedURL):
-            if resolvedURL.isIPFSScheme {
-              return handleIPFSSchemeURL(resolvedURL)
-            } else {
-              finishEditingAndSubmit(resolvedURL)
-              return true
-            }
-          case .none:
-            break
           }
+        case .none:
+          break
         }
-
-        // The user entered a URL, so use it.
-        // Determine if url navigation is done from favourites or bookmarks
-        // To handle bookmarklets properly
-        finishEditingAndSubmit(fixupURL, isUserDefinedURLNavigation: isUserDefinedURLNavigation)
-        return true
       }
     }
 
-    return false
+    // The user entered a URL, so use it.
+    // Determine if url navigation is done from favourites or bookmarks
+    // To handle bookmarklets properly
+    finishEditingAndSubmit(fixupURL, isUserDefinedURLNavigation: isUserDefinedURLNavigation)
+    return true
+  }
+
+  /// Handles displaying a Chromium web view for brave:// url that would display WebUI
+  func handleChromiumWebUIURL(_ url: URL) -> Bool {
+    let supportedPages = [
+      "flags",
+      "histograms",
+      "local-state",
+      "version",
```

Comment:
Most of the URLs from `brave://chrome-urls/` will also work :)
Not sure if we need them though, but if we do, we can just add them here

Reply:
Yeah technically we don't _need_ this whitelist at all if we can have `brave://sync-internals` load like normal

Reply:
The question is do we need other brave:// URLs available on iOS? I'd keep it limited if we don't need them. I'll need to look at sync-internals more closely, there is some data exporting functionalities that we may need to hide behind biometrics on Android.